sdk/azidentity/workload_identity.go

```diff
@@ -139,3 +159,115 @@ func (w *WorkloadIdentityCredential) getAssertion(context.Context) (string, erro
 	}
 	return w.assertion, nil
 }
+
+// aksTokenRequestPolicy redirects token requests to the AKS token endpoint, sending them via its
+// own HTTP client. It sends all other requests unchanged, via the pipeline's configured transport.
+type aksTokenRequestPolicy struct {
+	// c is configured for the AKS token endpoint
+	c *http.Client
```

Comment:
Will the request seen by this client still contain the correct headers such as user agent, etc?